src/models/StyledComponent.js

```diff
@@ -96,6 +96,19 @@ export default (ComponentStyle: Function, constructWithOptions: Function) => {
       }
     }
 
+    getThemeAndGeneratedClassName(props: any, fallbackTheme: any) {
+    // Props should take precedence over ThemeProvider, which should take precedence over
```

Comment:
As per your comment in the other thread:

Let’s make this a helper and reuse it across the Native component and withTheme, right?